geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateGatewayReceiverCommand.java

```diff
@@ -80,49 +81,75 @@ public Result createGatewayReceiver(@CliOption(key = {CliStrings.GROUP, CliStrin
       @CliOption(key = CliStrings.CREATE_GATEWAYRECEIVER__HOSTNAMEFORSENDERS,
           help = CliStrings.CREATE_GATEWAYRECEIVER__HOSTNAMEFORSENDERS__HELP) String hostnameForSenders,
 
-      @CliOption(key = {CliStrings.IFNOTEXISTS}, help = CliStrings.PUT__PUTIFNOTEXISTS__HELP,
+      @CliOption(key = CliStrings.IFNOTEXISTS, help = CliStrings.IFNOTEXISTS_HELP,
           specifiedDefaultValue = "true", unspecifiedDefaultValue = "false") Boolean ifNotExists) {
 
-    GatewayReceiverFunctionArgs gatewayReceiverFunctionArgs = new GatewayReceiverFunctionArgs(
-        manualStart, startPort, endPort, bindAddress, socketBufferSize, maximumTimeBetweenPings,
-        gatewayTransportFilters, hostnameForSenders, ifNotExists);
+    CacheConfig.GatewayReceiver configuration =
+        buildConfiguration(manualStart, startPort, endPort, bindAddress, maximumTimeBetweenPings,
+            socketBufferSize, gatewayTransportFilters, hostnameForSenders);
+
+    GatewayReceiverFunctionArgs gatewayReceiverFunctionArgs =
+        new GatewayReceiverFunctionArgs(configuration, ifNotExists);
 
     Set<DistributedMember> membersToCreateGatewayReceiverOn = getMembers(onGroups, onMember);
 
     List<CliFunctionResult> gatewayReceiverCreateResults =
         executeAndGetFunctionResult(GatewayReceiverCreateFunction.INSTANCE,
             gatewayReceiverFunctionArgs, membersToCreateGatewayReceiverOn);
 
-    CommandResult result = ResultBuilder.buildResult(gatewayReceiverCreateResults);
-
-    XmlEntity xmlEntity = findXmlEntity(gatewayReceiverCreateResults);
-    // no xml needs to be updated, simply return
-    if (xmlEntity == null) {
-      return result;
-    }
+    ResultModel result = ResultModel.createMemberStatusResult(gatewayReceiverCreateResults);
 
-    boolean allSuccessful = gatewayReceiverCreateResults.stream()
-        .map(CliFunctionResult::isSuccessful).reduce(true, (x, y) -> x && y);
-    if (!allSuccessful) {
+    boolean anySuccessful = gatewayReceiverCreateResults.stream()
```

Comment:
Is this block still necessary? I think ResultModel.ceateMemberStatusResult already sets the flag.

Reply:
Oh, duh.  Good catch.